lib/membrane/video_compositor/video_compositor.ex

```diff
@@ -0,0 +1,431 @@
+          )
+        end
+
+        {[notify_parent: {:vc_request_response, request_body, response, State.ctx(state)}], state}
+
+      {:error, err} ->
+        Membrane.Logger.error("Request: #{request_body} failed. Error: #{err}.")
+        {[], state}
+    end
+  end
+
+  @impl true
+  def handle_parent_notification(_notification, _ctx, state = %State{}) do
+    {[], state}
+  end
+
+  @impl true
+  def handle_child_notification(
+        {:new_rtp_stream, ssrc, _pt, _ext},
+        {:rtp_receiver, pad_id},
+        _ctx,
+        state = %State{}
+      ) do
+    %OutputState{resolution: %Resolution{width: output_width, height: output_height}} =
+      state.outputs
+      |> Enum.find(fn %OutputState{pad_ref: Pad.ref(:output, id)} -> pad_id == id end)
+
+    output_stream_format = %Membrane.H264{
+      framerate: {state.framerate, 1},
+      alignment: :nalu,
+      stream_structure: :annexb,
+      width: output_width,
+      height: output_height
+    }
+
+    spec =
+      get_child({:rtp_receiver, pad_id})
+      |> via_out(Pad.ref(:output, ssrc), options: [depayloader: RTP.H264.Depayloader])
+      |> child({:output_processor, pad_id}, %Membrane.VideoCompositor.OutputProcessor{
+        output_stream_format: output_stream_format
+      })
+      |> bin_output(Pad.ref(:output, pad_id))
+
+    {[spec: spec], state}
+  end
+
+  @impl true
+  def handle_child_notification(_msg, _child, _ctx, state) do
+    {[], state}
+  end
+
+  @spec start_vc_server(port_number()) :: :ok
+  defp start_vc_server(vc_port) do
+    architecture = system_architecture() |> Atom.to_string()
+
+    vc_app_path =
+      File.cwd!()
+      |> Path.join("video_compositor_app/#{architecture}/video_compositor/video_compositor")
+
+    spawn(fn ->
+      Rambo.run(vc_app_path, [], env: %{"MEMBRANE_VIDEO_COMPOSITOR_API_PORT" => "#{vc_port}"})
+    end)
+
+    started? =
+      0..50
+      |> Enum.reduce_while(false, fn _i, _acc ->
+        sleep_time_ms = 100
+        :timer.sleep(sleep_time_ms)
+
+        case VcReq.send_custom_request(%{}, vc_port) do
+          {:ok, _} -> {:halt, true}
+          {:error, _} -> {:cont, false}
+        end
+      end)
+
+    unless started? do
+      raise "Failed to startup and connect to VideoCompositor server."
+    end
+
+    :ok
+  end
+
+  @spec system_architecture() :: :darwin_aarch64 | :darwin_x86_64 | :linux_x86_64
+  defp system_architecture() do
+    case :os.type() do
+      {:unix, :darwin} ->
+        system_architecture = :erlang.system_info(:system_architecture) |> to_string()
+
+        cond do
+          Regex.match?(~r/aarch64/, system_architecture) ->
+            :darwin_aarch64
+
+          Regex.match?(~r/x86_64/, system_architecture) ->
+            :darwin_x86_64
+
+          true ->
+            raise "Unsupported system architecture: #{system_architecture}"
+        end
+
+      {:unix, :linux} ->
+        :linux_x86_64
+
+      os_type ->
+        raise "Unsupported os type: #{os_type}"
+    end
+  end
```

Comment:
1) This same issue as in `Mix.Tasks.DownloadCompositor`
2) You have equal private function in `Mix.Tasks.DownloadCompositor`. Don't repeat yourself and make this function public, maybe extract it to another module